x-pack/plugins/security_solution/server/search_strategy/security_solution/factory/last_first_seen/query.first_or_last_seen.dsl.ts

```diff
@@ -5,21 +5,25 @@
  * 2.0.
  */
 
-import { HostFirstLastSeenRequestOptions } from '../../../../../../common/search_strategy/security_solution/hosts';
+import { isEmpty } from 'lodash/fp';
+import { FirstLastSeenRequestOptions } from '../../../../../common/search_strategy/security_solution/first_last_seen';
 
-export const buildFirstOrLastSeenHostQuery = ({
-  hostName,
+export const buildFirstOrLastSeenQuery = ({
+  field,
+  value,
   defaultIndex,
+  docValueFields,
   order,
-}: HostFirstLastSeenRequestOptions) => {
-  const filter = [{ term: { 'host.name': hostName } }];
+}: FirstLastSeenRequestOptions) => {
+  const filter = [{ term: { [field]: value } }];
 
   const dslQuery = {
     allow_no_indices: true,
     index: defaultIndex,
     ignore_unavailable: true,
     track_total_hits: false,
     body: {
+      ...(!isEmpty(docValueFields) ? { docvalue_fields: docValueFields } : {}),
```

Comment:
Do we really need the `docvalue_fields` in this query? we only use the `@timestamp` in the hits from the response

Reply:
fixed!